skimage/transform/tests/test_geometric.py

```diff
@@ -630,22 +651,34 @@ def _assert_least_squares(tf, src, dst):
             assert new_ssq > baseline
 
 
-def test_estimate_affine_3d():
+@pytest.mark.parametrize('array_like_input', [False, True])
+def test_estimate_affine_3d(array_like_input):
     ndim = 3
     src = np.random.random((25, ndim)) * 2 ** np.arange(7, 7 + ndim)
-    matrix = np.array(
-        [[4.8, 0.1, 0.2, 25],
-         [0.0, 1.0, 0.1, 30],
-         [0.0, 0.0, 1.0, -2],
-         [0.0, 0.0, 0.0, 1.]]
-    )
+
+    matrix = [
+        [4.8, 0.1, 0.2, 25],
+        [0.0, 1.0, 0.1, 30],
+        [0.0, 0.0, 1.0, -2],
+        [0.0, 0.0, 0.0, 1.]
+    ]
+    if not array_like_input:
+        matrix = np.asarray(matrix)
+    else:
+        # list of lists for matrix and src coords
+        src = [list(c) for c in src]
```

Comment:
```suggestion
    matrix = np.array([
        [4.8, 0.1, 0.2, 25],
        [0.0, 1.0, 0.1, 30],
        [0.0, 0.0, 1.0, -2],
        [0.0, 0.0, 0.0, 1.]
    ])
    
    if array_like_input:
        # list of lists for matrix and src coords
        src = [list(c) for c in src]
        matrix = [list(c) for c in matrix]
```
(code is correct but too unnatural)